packages/instant/src/containers/selected_asset_buy_order_state_buttons.ts

```diff
@@ -78,21 +87,26 @@ const mapDispatchToProps = (
     dispatch: Dispatch<Action>,
     ownProps: SelectedAssetBuyOrderStateButtons,
 ): ConnectedDispatch => ({
-    onValidationPending: (buyQuote: BuyQuote) => {
-        dispatch(actions.setBuyOrderStateValidating());
+    onValidationPending: (swapQuote: MarketBuySwapQuote) => {
+        dispatch(actions.setSwapOrderStateValidating());
     },
-    onBuyProcessing: (buyQuote: BuyQuote, txHash: string, startTimeUnix: number, expectedEndTimeUnix: number) => {
-        dispatch(actions.setBuyOrderStateProcessing(txHash, startTimeUnix, expectedEndTimeUnix));
+    onBuyProcessing: (
+        swapQuote: MarketBuySwapQuote,
+        txHash: string,
+        startTimeUnix: number,
+        expectedEndTimeUnix: number,
+    ) => {
+        dispatch(actions.setSwapOrderStateProcessing(txHash, startTimeUnix, expectedEndTimeUnix));
     },
-    onBuySuccess: (buyQuote: BuyQuote, txHash: string) => dispatch(actions.setBuyOrderStateSuccess(txHash)),
-    onBuyFailure: (buyQuote: BuyQuote, txHash: string) => dispatch(actions.setBuyOrderStateFailure(txHash)),
+    onBuySuccess: (swapQuote: MarketBuySwapQuote, txHash: string) => dispatch(actions.setSwapOrderStateSuccess(txHash)),
+    onBuyFailure: (swapQuote: MarketBuySwapQuote, txHash: string) => dispatch(actions.setSwapOrderStateFailure(txHash)),
```

Comment:
A good follow-up here would be to show an error specific to the protocol fee failure edge cases.

Reply:
I'll tentatively add a TODO for this, one large feedback at EthWaterloo is we need richer errors.